plugins/webp-uploads/hooks.php

```diff
@@ -803,3 +803,63 @@ function webp_uploads_enable_additional_mime_type_support_for_all_sizes( array $
 	return $allowed_sizes;
 }
 add_filter( 'webp_uploads_image_sizes_with_additional_mime_type_support', 'webp_uploads_enable_additional_mime_type_support_for_all_sizes' );
+
+/**
+ * Converts palette PNG images to true color PNG images.
+ *
+ * GD cannot convert palette-based PNG to WebP/AVIF formats, causing conversion failures.
+ * This function detects and converts palette PNG to truecolor during upload.
+ *
+ * @since n.e.x.t
+ *
+ * @param array<string, mixed> $file The uploaded file data.
+ * @return array<string, mixed> The modified file data.
+ */
+function webp_uploads_convert_palette_png_to_truecolor( array $file ): array {
```

Comment:
Since WordPress plugins can supply any kind of value in filters, this should have better safeguards to prevent a PHP fatal error:

```suggestion
 * @param array<string, mixed>|mixed $file The uploaded file data.
 * @return array<string, mixed> The modified file data.
 */
function webp_uploads_convert_palette_png_to_truecolor( $file ): array {
	if ( ! is_array( $file ) ) {
		$file = array();
	}
```